### PR TITLE
Remove value tuples from public TransformCatalog APIs

### LINKED ISSUE #2581: Remove value tuples from TransformsCatalog methods
Many of the methods in TransformsCatalog create estimators using a params array of (string outputColumnName, string inputColumnName). As mentioned in #2501, we should change all these to not use value tuples.

### PR DESCRIPTION
**Pull request by yaeldekel**
Fixes #2581 .

### CODE REVIEW

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 1]
```diff
@@ -2,12 +2,36 @@
...
 // See the LICENSE file in the project root for more information.
 
+using System.Linq;
 using Microsoft.Data.DataView;
 using Microsoft.ML.Data;
 using Microsoft.ML.Transforms;
 
 namespace Microsoft.ML
 {
+    public sealed class SimpleColumnInfo
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: >SimpleColumnInfo [](start = 24, length = 16)

I think @rogancarr has issue where he don't like ColumnInfo name. [https://github.com/dotnet/machinelearning/issues/2554](https://github.com/dotnet/machinelearning/issues/2554)
I wonder how much he would hate this one.

`SimpleMapping`,`SimplePair`?

Comment (rogancarr) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: No strong opinion, I just think that we should have a conversation about naming these things :)

---
In reply to: [258285683](https://github.com/dotnet/machinelearning/pull/2641#discussion_r258285683) [](ancestors = 258285683)

Comment (yaeldekel) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: How about `ColumnNamePair` or something similar? I think something containing 'mapping' or 'pair' would be too general, wouldn't it?

Or I could leave it as `SimpleColumnInfo` and add a comment about it in the issue so we can change it when we find a good name for `ColumnInfo`?

---
In reply to: [258297894](https://github.com/dotnet/machinelearning/pull/2641#discussion_r258297894) [](ancestors = 258297894,258285683)

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: I added mention of this column to #2554.
Let's merge this one, and figure out proper naming in that issue.

**Pull request merged**